### PR TITLE
Use defaults in iCE40 binding

### PR DESCRIPTION
**Pull request by benediktibk**
Replace the DT_INST_PROP_OR statements with defaults in the devicetree binding of the iCE40.

### CODE REVIEW

**Review by josuah — DISMISSED** [review]
Thank you for porting these changes from #80854 !

**Review by cfriedt — DISMISSED** [review]

**Review by tgorochowik — DISMISSED** [review]

File: dts/bindings/fpga/lattice,ice40-fpga.yaml
Comment (henrikbrixandersen) [inline dts/bindings/fpga/lattice,ice40-fpga.yaml:1]: Default values in bindings need to be explained/justified in the description (please see https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#rules-for-default-values).

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: dts/bindings/fpga/lattice,ice40-fpga.yaml
Comment (benediktibk) [reply dts/bindings/fpga/lattice,ice40-fpga.yaml:1]: done

**Review by henrikbrixandersen — APPROVED** [review]
Thanks.

**Review by josuah — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Review by tgorochowik — APPROVED** [review]

**Pull request merged**